### PR TITLE
[FLINK-25874] loose PyFlink dependency of 'python-dateutil' to '>=2.8.0,<3'

### PR DESCRIPTION
**Pull request by martin-liu**
## What is the purpose of the change

[FLINK-25874](https://issues.apache.org/jira/browse/FLINK-25874) reported that PyFlink dependency of `python-dateutil==2.8.0` is [very old](https://github.com/dateutil/dateutil/releases/tag/2.8.0) and conflict with some python packages like [great_expectations](https://github.com/great-expectations/great_expectations/).
This PR looses the dependency. 

## Brief change log

* loose PyFlink dependency of 'python-dateutil' to '>=2.8.0,<3'

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by martin-liu** [comment]
@HuangXingBo Can you take a look? Thanks!

File: flink-python/dev/dev-requirements.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ wheel
 apache-beam==2.27.0
 cython==0.29.24
 py4j==0.10.9.3
-python-dateutil==2.8.0
+python-dateutil==2.8.2
```
Comment (dianfu) [inline flink-python/dev/dev-requirements.txt:20]: Keep it consistent with setup.py?
```suggestion
python-dateutil>=2.8.0,<3
```

**Review by dianfu — COMMENTED** [review]
@martin-liu Thanks for the PR. Have left a minor comment.

**Comment by martin-liu** [comment]
@dianfu Good catch, accepted the suggestion, thank you very much!

**Comment by martin-liu** [comment]
@flinkbot run azure

**Comment by martin-liu** [comment]
@dianfu The CI check is finished successfully, anything else do we need?

**Comment by martin-liu** [comment]
@HuangXingBo @dianfu can you take a look on this?

**Comment by martin-liu** [comment]
@dianfu This PR is pending for almost 3 months, can we proceed it if there's no more concern?

**Comment by dianfu** [comment]
Very sorry for the late response. LGTM.

**Pull request closed without merging**